### PR TITLE
feat: Add charge prices table

### PR DESCRIPTION
**Pull request by ShaBloze**
## Description

Add charge prices table in drawer with data.

## References

- #1688
https://app.zenhub.com/workspaces/team-volt-60a61059028fd7000e49956e/issues/energinet-datahub/geh-charges/1688

### CODE REVIEW

File: libs/dh/shared/ui-paginator/src/lib/dh-shared-ui-paginator.component.ts [hunk 1]
```diff
@@ -63,6 +73,11 @@ export class DhSharedUiPaginatorComponent implements OnDestroy {
     this.destroy$.complete();
   }
 
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  handlePageEvent(event: any) {
```
Comment (ManBearTM) [inline libs/dh/shared/ui-paginator/src/lib/dh-shared-ui-paginator.component.ts:77]: I think you should be able to type this as `PageEvent`

File: libs/dh/charges/feature-prices/src/lib/drawer/price-tab/dh-charges-charge-prices-tab.component.html [hunk 2]
```diff
@@ -14,8 +14,124 @@
...
+        <ng-container matColumnDef="fromDateTime">
+          <th *matHeaderCellDef mat-header-cell mat-sort-header>
+            {{ t("priceTab.fromDateTime") }}
+          </th>
+          <td mat-cell *matCellDef="let element">
+            {{ element.fromDateTime | dhDate }}
+          </td>
+        </ng-container>
+
+        <!-- To date time column -->
```
Comment (HenrikSommer) [inline libs/dh/charges/feature-prices/src/lib/drawer/price-tab/dh-charges-charge-prices-tab.component.html:73]:
```suggestion
        <!-- Time column -->
```

File: libs/dh/charges/feature-prices/src/lib/drawer/price-tab/dh-charges-charge-prices-tab.component.html [hunk 3]
```diff
@@ -14,8 +14,124 @@
...
+          <th *matHeaderCellDef mat-header-cell mat-sort-header disabled>
+            {{ t("priceTab.time") }}
+          </th>
+          <td mat-cell *matCellDef="let element">
+            {{ this.getTime(element.fromDateTime) }} -
+            {{ this.getTime(element.toDateTime) }}
+          </td>
+        </ng-container>
+
+        <!-- valid to date column -->
```
Comment (HenrikSommer) [inline libs/dh/charges/feature-prices/src/lib/drawer/price-tab/dh-charges-charge-prices-tab.component.html:84]:
```suggestion
        <!-- Valid to date column -->
```

Comment (HenrikSommer) [inline libs/dh/charges/feature-prices/src/lib/drawer/price-tab/dh-charges-charge-prices-tab.component.html:84]:
```suggestion
        <!-- Price column -->
```

File: libs/dh/charges/feature-prices/src/lib/drawer/price-tab/dh-charges-charge-prices-tab.component.ts [hunk 4]
```diff
@@ -14,24 +14,236 @@
...
   dateRangeChanged(dateRange: DatePickerData) {
-    console.log(dateRange);
+    this.searchCriteria.fromDateTime = zonedTimeToUtc(
+      dateRange.startDate,
+      this.localTimeZone
+    ).toISOString();
+    this.searchCriteria.toDateTime = zonedTimeToUtc(
+      dateRange.endDate,
+      this.localTimeZone
+    ).toISOString();
```
Comment (HenrikSommer) [inline libs/dh/charges/feature-prices/src/lib/drawer/price-tab/dh-charges-charge-prices-tab.component.ts:157]: Lines 150 - 157 are duplicated in 172 - 179. I suggest we move the code to a new method.

File: libs/dh/shared/domain/src/lib/generated/v1/model/models.ts [hunk 5]
```diff
@@ -50,8 +54,8 @@ export * from './process-status';
 export * from './product-id';
 export * from './reading-occurrence';
 export * from './resolution';
-export * from './search-criteria-v1-dto';
 export * from './settlement-method';
+export * from './sort-column-name';
```
Comment (HenrikSommer) [inline libs/dh/shared/domain/src/lib/generated/v1/model/models.ts:58]: That's very generic - but the class isn't.

**Review by HenrikSommer — APPROVED** [review]

**Pull request merged**